File: r/2015/day08b/day08b.r
#!/usr/bin/env Rscript

usage <- function() {
    args <- commandArgs()
    progname <- sub("^--file=", "", args[grep("^--file=", args)])
    cat("usage:", progname, "<input file>\n")
    quit(status = 1)
}

process <- function(filename) {
    result <- 0
    lines <- suppressWarnings(readLines(filename))
    for ( line in lines ) {
        codeLen <- nchar(line)
        encLen <- 0
        for ( i in 1:nchar(line) ) {
            if ( substr(line, i, i) %in% c('\\', '"') ) {
                encLen <- encLen + 2
            } else {
                encLen <- encLen + 1
            }
        }
        result <- result + 2 + (encLen - codeLen)
    }
    return(result)
}

main <- function() {
    args <- commandArgs(trailingOnly = TRUE)
    if ( length(args) < 1 ) {
        usage()
    }
    filename <- args[1]
    result <- process(filename)
    cat("result =", result, "\n")
}

main()

File: dart/2015/day08b/day08b.dart
#!/usr/bin/env dart

import 'dart:io';

void usage(String progName) {
    print('usage: $progName <input file>');
    exit(1);
}

int process(String filename) {
    int result = 0;

    final lines = File(filename).readAsLinesSync();
    for ( final line in lines ) {
        final codeLen = line.length;
        int encLen = 0;
        for ( int i = 0; i < line.length; i++ ) {
            final ch = line.substring(i, i + 1);
            switch ( ch ) {
            case '\\':
            case '\"':
                encLen += 2;
                break;
            default:
                encLen += 1;
            }
        }
        result += 2 + (encLen - codeLen);
    }
    return result;
}

void main(List<String> args) {
    if ( args.isEmpty ) {
        usage(Platform.script.pathSegments.last);
    }
    final filename = args[0];
    final result = process(filename);
    print('result = $result');
}
